### ISSUE TITLE
Save average document length after fitting

### ISSUE THREAD

**Issue by dunefox** [arosh/BM25Transformer#1]
I think [avgdl](https://github.com/arosh/BM25Transformer/blob/master/bm25.py#L76) should be saved as an attribute after fitting so it's not estimated again if transform is called for one document instead of the 'training' corpus.

So

```python
fit(X).transform(X)
```

makes sense because all documents in X use the same avgdl but

```python
fit(X).transform(X)
transform(other_document)
```

then estimates avgdl for this document alone again.

**Comment by arosh** [comment]
You are completely right. It is terrible that this issue has been overlooked for more than several years...
